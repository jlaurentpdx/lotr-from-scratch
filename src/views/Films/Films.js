import './Films.css';

export default function Films() {
  return <div>Films page</div>;
}
