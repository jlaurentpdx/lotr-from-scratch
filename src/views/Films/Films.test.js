import { render, screen } from '@testing-library/react';
import Films from './Films';

test('Films should render a list of films with box office total and nominations', async () => {
  const { container } = render(<Films />);

  await screen.findByText('The Return of the King');
  await screen.findByText('Box Office Total: $1146M');
  await screen.findByText('Nominated for 11 Academy Awards');

  expect(container).toMatchSnapshot();
});
