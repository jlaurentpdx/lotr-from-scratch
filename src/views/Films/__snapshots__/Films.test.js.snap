// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films should render a list of films with box office total and nominations 1`] = `
<div>
  <div
    class="films-page"
  >
    <h2>
      Films
    </h2>
    <div
      class="film-list"
    >
      <div>
        <h3>
          The Two Towers
        </h3>
        <p>
          Box Office Total: $
          948
          M
        </p>
        <p>
          Nominated for 
          23
           Academy Awards
        </p>
      </div>
      <div>
        <h3>
          The Return of the King
        </h3>
        <p>
          Box Office Total: $
          1146
          M
        </p>
        <p>
          Nominated for 
          11
           Academy Awards
        </p>
      </div>
      <div>
        <h3>
          The Fellowship of the Ring
        </h3>
        <p>
          Box Office Total: $
          898
          M
        </p>
        <p>
          Nominated for 
          13
           Academy Awards
        </p>
      </div>
      <div>
        <h3>
          The Desolation of Smaug
        </h3>
        <p>
          Box Office Total: $
          959
          M
        </p>
        <p>
          Nominated for 
          3
           Academy Awards
        </p>
      </div>
      <div>
        <h3>
          The Battle of the Five Armies
        </h3>
        <p>
          Box Office Total: $
          962
          M
        </p>
        <p>
          Nominated for 
          1
           Academy Awards
        </p>
      </div>
      <div>
        <h3>
          The Unexpected Journey
        </h3>
        <p>
          Box Office Total: $
          1000
          M
        </p>
        <p>
          Nominated for 
          3
           Academy Awards
        </p>
      </div>
    </div>
  </div>
</div>
`;
